src/pages/Group/Group.js

```diff
@@ -1,5 +1,392 @@
+import React, { useState, Image } from 'react';
+import './Group.css'
+import { useNavigate } from 'react-router-dom';
+import Header from '../../components/Header/Header';
+import Card from '../../components/Card/Card';
+import Button from '../../components/Button/Button';
+import FilterButton from '../../components/FilterButton/FilterButton';
+import { Icon } from '@iconify/react';
+
 function Group() {
+    const navigate = useNavigate();
+    const [filterBtnStyles, setFilterBtnStyles] = useState(Array.from( {length: 5}, () => false));
+    const [cuisineClicked, setCuisineClicked] = useState(Array.from( {length: 6}, () => false));
+    const [priceClicked, setPriceClicked] = useState(Array.from( {length: 3}, () => false ));
+    const [dietClicked, setDietClicked] = useState(Array.from( {length: 6}, () => false));
+    const [mealClicked, setMealclicked] = useState(Array.from( {length: 6}, () => false));
+
+    const handleSingleBtnColor = (arr, setArr, index) => {
+        const newBtnStyles = arr.map((value, currIndex) => {
```

Comment:
nit: if you're using a ternary you can simplify the code into arr.map((val, currIndex) => currIndex===index)